sdks/go/sdk.go

```diff
@@ -76,3 +87,43 @@ func (s *SDK) Health() error {
 func (s *SDK) GameServer() (*sdk.GameServer, error) {
 	return s.client.GetGameServer(s.ctx, &sdk.Empty{})
 }
+
+// WatchGameServer asynchronously calls the given GameServerCallback with the current GameServer
+// configuration when the backing GameServer configuration is updated.
+// This function can be called multiple times to add more than one GameServerCallback.
+func (s *SDK) WatchGameServer(f GameServerCallback) error {
+	s.watchCallbacks = append(s.watchCallbacks, f)
+	var err error
+
+	s.watchOnce.Do(func() {
```

Comment:
@enocom I've love to get your take on this approach. WDYT?

Reply:
Instead of using a `sync.Once`, it might be easier to have a `RegisterCallback` method to register a new callback. Then `WatchGameServer` can run forever invoking the callbacks. 

Presently, there's a data race between the call to `append` above and reading the callbacks in the goroutine with `range` below. If you split out the two concerns, you can lock around the slice of callbacks, taking out the lock when appending or reading from the slice, and then releasing it afterwords.

Be careful about locking around the callback, though, as there is a chance the callback might never finish. You might consider passing a `content.Context` into the callback to provide a mechanism for cancelling the work and releasing the lock in failure cases.

Reply:
Reading all this, and your comments I think I've got too fancy with this. 

I think I'll take an approach that is similar to what I did with the C++ (although non-blocking). Each call to `WatchGameServer` is it's own call to the gRPC service, with it's own *singular* function callback - that will make things far simpler to run and test, rather than managing sets of function callbacks and the potential issues therein.